lib/Dialect/Ibis/Transforms/IbisContainersToHW.cpp

```diff
@@ -34,41 +35,54 @@ struct ContainerPortInfo {
   // A mapping between the port name and the port op within the container.
   llvm::DenseMap<StringAttr, OutputPortOp> opOutputs;
 
+  // A mapping between port symbols and their corresponding port name.
+  llvm::DenseMap<StringAttr, StringAttr> portSymbolsToPortName;
+
   ContainerPortInfo() = default;
   ContainerPortInfo(ContainerOp container) {
     SmallVector<hw::PortInfo, 4> inputs, outputs;
+    auto *ctx = container.getContext();
 
-    // Copies all attributes from a port, except for the port symbol and type.
-    auto copyPortAttrs = [](auto port) {
+    // Copies all attributes from a port, except for the port symbol, name, and
+    // type.
+    auto copyPortAttrs = [ctx](auto port) {
       llvm::DenseSet<StringAttr> elidedAttrs;
       elidedAttrs.insert(port.getInnerSymAttrName());
       elidedAttrs.insert(port.getTypeAttrName());
+      elidedAttrs.insert(port.getNameAttrName());
       llvm::SmallVector<NamedAttribute> attrs;
       for (NamedAttribute namedAttr : port->getAttrs()) {
         if (elidedAttrs.contains(namedAttr.getName()))
           continue;
         attrs.push_back(namedAttr);
       }
-      return DictionaryAttr::get(port.getContext(), attrs);
+      return DictionaryAttr::get(ctx, attrs);
     };
 
-    // Gather in and output port ops.
+    // Gather in and output port ops to define the hw.module interface. Here, we
+    // also perform uniqueing of the port names.
+    Namespace portNs;
     for (auto input : container.getBodyBlock()->getOps<InputPortOp>()) {
-      opInputs[input.getInnerSym().getSymName()] = input;
-
+      auto uniquePortName =
```

Comment:
I may be wrong about this, but I don't think port names are required to be unique in the `hw.module` IR. IIRC ExportVerilog will take care of uniquing port names. In the IR, they are pretty much only name hints -- only order matters.